src/server/middleware/security/idtoken.go

```diff
@@ -40,6 +40,9 @@ func (i *idToken) Generate(req *http.Request) security.Context {
 	if !strings.HasPrefix(req.URL.Path, "/api") {
 		return nil
 	}
+	if strings.Contains(req.URL.Path, "ping") {
+		return nil
+        }
```

Comment:
Won't this permit all requests with `ping` in the name to bypass OIDC auth? Say, I create an account/namespace/... called `ping` - then authentication's off.

(I know Harbor mostly uses numerical IDs in URLs, but that might change later on.)

```suggestion
      # Don't complain on healthchecks without OIDC login
  	if req.URL.path == "/api/v2.0/ping") {
		return nil
        }
```

Reply:
That's also fine. I added the ping contains in case the path ever changed, but I suppose this change can be updated then too.